hnn_core/network.py

```diff
@@ -135,34 +136,49 @@ class Network(object):
     Parameters
     ----------
     params : dict
-        The parameters
+        The parameters to use for constructing the network.
 
     Attributes
     ----------
     params : dict
-        The parameters
+        The parameters of the network
     cellname_list : list
         The names of real cell types in the network (e.g. 'L2_basket')
     feedname_list : list
         The names of external drivers ('feeds') to the network (e.g. 'evdist1')
     gid_dict : dict
-        Dictionary with keys 'evprox1', 'evdist1' etc.
-        containing the range of Cell IDs of different cell
-        (or input) types.
+        A dictionary of unique identifiers of each real and artificial cell
+        in the network. Every cell type is represented by a key read from
+        cellname_list, followed by entries read from feedname_list. The value
+        of each key is a range of ints, one for each cell in given category.
+        Examples: 'L2_basal': range(0, 270), 'evdist1': range(272, 542), etc
+    pos_dict : dict
+        Dictionary containing the coordinate positions of all cells.
+        Keys are 'L2_pyramidal', 'L5_pyramidal', 'L2_basket', 'L5_basket',
+        'common', or any of the elements of the cellname or feedname lists.
     spikes : Spikes
         An instance of the Spikes object.
     """
 
     def __init__(self, params):
         # Save the parameters used to create the Network
         self.params = params
+        # Initialise a dictionary of cell ID's, which get used when the
+        # network is constructed ('built') in NetworkBuilder
+        # We want it to remain in each Network object, so that the user can
+        # interrogate a built and simulated net. In addition, Spikes are
+        # attached to a Network during simulation---Network is the natural
+        # place to keep this information
+        # XXX rename gid_dict in future
```

Comment:
let's just do it in this PR. Seems straightforward. `gid_ranges` ?

Reply:
done